lib/spack/llnl/util/filesystem.py

```diff
@@ -1856,7 +1856,12 @@ def find_libraries(libraries, root, shared=True, recursive=False):
         raise TypeError(message)
 
     # Construct the right suffix for the library
-    if shared:
+    if is_windows:
+        if shared:
+            suffixes = ['dll']
+        else:
+            suffixes = ['lib']
```

Comment:
I think this is causing most parameterizations of `test_searching_order` to fail because we don't properly set up the test on windows to create fake library files with the suffixes that one would expect there: would you be able to update that test?

Reply:
I can certainly look at it!